index.d.ts

```diff
@@ -428,10 +428,11 @@ declare namespace Moleculer {
 
 	type ActionCacheEnabledFuncType = (ctx: Context<any, any>) => boolean;
 
-	interface ActionCacheOptions {
+	interface ActionCacheOptions<P = Record<string, any>, M = any, K = any[]|null> {
```

Comment:
The `keys` property is indicated as `Array<string> | undefined` but the `K` generic here (as well as in `CacherKeygenFuncType` defaults to `any[]`.  Either `keys` needs to be widened (which I don't think is a good idea) or the `K` generic should be eliminated and the `defaultKeys` parameter set to `defaultKeys?: string[]`.

Although the jsdocs indicate that `keys` can be `null`, I don't actually see that.  Based on the types below, keys is either `undefined` or `string[]` so I think the `null` would be incorrect which is resolved with my suggestion above.